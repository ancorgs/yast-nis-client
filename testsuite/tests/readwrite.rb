# encoding: utf-8

# Module:
#   NIS client configuration
#
# Summary:
#   Testsuite
#
# Authors:
#   Martin Vidner <mvidner@suse.cz>
#
# $Id$
module Yast
  class ReadwriteClient < Client
    def main
      # testedfiles: Nis.ycp Autologin.ycp Service.ycp Report.ycp Testsuite.ycp Nsswitch.ycp

      Yast.include self, "testsuite.rb"

      @READ_INIT = { "target" => { "size" => 0 } }
      @EXEC_INIT = { "target" => { "bash_output" => {} } }

      TESTSUITE_INIT([@READ_INIT, {}, @EXEC_INIT], nil)

      Yast.import "Pkg" # override
      Yast.import "Nis"


      @READ = {
        # Runlevel:
        "init"      => {
          "scripts" => {
            "exists"   => true,
            "runlevel" => {
              "rpcbind" => { "start" => ["3", "5"], "stop" => ["3", "5"] },
              "ypbind"  => { "start" => ["3", "5"], "stop" => ["3", "5"] },
              "autofs"  => { "start" => ["3", "5"], "stop" => ["3", "5"] }
            },
            # their contents is not important for ServiceAdjust
            "comment"  => {
              "rpcbind" => {},
              "ypbind"  => {},
              "autofs"  => {}
            }
          }
        },
        # Nis itself:
        "etc"       => {
          "yp_conf"       => {
            "servers"          => ["10.20.30.40", "10.20.30.80"],
            "slp"              => { "slpdomain" => true },
            "defaultbroadcast" => false,
            "domainservers"    => { "otherdomain" => ["1.2.3.4"] },
            "broadcast"        => { "otherdomain" => true }
          },
          #	    "defaultdomain": "mydomain",
          "nsswitch_conf" => {
            "passwd"    => "compat",
            "group"     => "compat",
            "hosts"     => "files dns6",
            "automount" => "files",
            "services"  => "files",
            "netgroup"  => "files",
            "aliases"   => "files"
          }
        },
        "sysconfig" => {
          "ypbind"         => {
            "YPBIND_LOCAL_ONLY"    => "no",
            "YPBIND_BROADCAST"     => "no",
            "YPBIND_BROKEN_SERVER" => "no",
            "YPBIND_OPTIONS"       => ""
          },
          "network"        => { "config" => { "NETCONFIG_NIS_POLICY" => "" } },
          "displaymanager" => {
            "DISPLAYMANAGER"                     => "kdm",
            "DISPLAYMANAGER_AUTOLOGIN"           => "no",
            "DISPLAYMANAGER_PASSWORD_LESS_LOGIN" => "no"
          }
        },
        "target"    => {
          "size" => 0,
          # FileUtils::Exists
          "stat" => { 1 => 2 }
        }
      }

      @WRITE = {}

      @EXECUTE = {
        "target" =>
          # /etc/yp.conf.sv
          {
            # ok if used both for `domainname` and `rcypbind start`
            "bash_output" => {
              "exit"   => 0,
              "stdout" => "mydomain\n",
              "stderr" => ""
            },
            "remove"      => true
          }
      }

      DUMP("no policy")
      TEST(lambda { Nis.Read }, [@READ, @WRITE, @EXECUTE], nil)
      TEST(lambda { Nis.Write }, [@READ, @WRITE, @EXECUTE], nil)

      DUMP("auto policy")
      Ops.set(
        @READ,
        ["sysconfig", "network", "config", "NETCONFIG_NIS_POLICY"],
        "auto"
      )
      TEST(lambda { Nis.Read }, [@READ, @WRITE, @EXECUTE], nil)
      TEST(lambda { Nis.Touch(true) }, [], nil)
      TEST(lambda { Nis.Write }, [@READ, @WRITE, @EXECUTE], nil)

      nil
    end
  end
end

Yast::ReadwriteClient.new.main
