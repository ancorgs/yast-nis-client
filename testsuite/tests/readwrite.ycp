/**
 * Module:
 *   NIS client configuration
 *
 * Summary:
 *   Testsuite
 *
 * Authors:
 *   Martin Vidner <mvidner@suse.cz>
 *
 * $Id$
 */
{
    include "testsuite.ycp";
    import "Pkg"; // override
    import "Nis";

    map dhcp_yes = $[
	"DHCLIENT_MODIFY_NIS_CONF": "yes",
	"DHCLIENT_SET_DOMAINNAME": "yes",
	];
    map dhcp_no = $[
	"DHCLIENT_MODIFY_NIS_CONF": "no",
	"DHCLIENT_SET_DOMAINNAME": "no",
	];

    map READ = $[
	// Runlevel:
	"init": $[
	    "scripts": $[
		"exists": true,
		"runlevel": $[
		    "portmap": $[
			"start": [ "3", "5"],
			"stop":	 [ "3", "5"],
			],
		    "ypbind": $[
			"start": [ "3", "5"],
			"stop":	 [ "3", "5"],
			],
		    "autofs": $[
			"start": [ "3", "5"],
			"stop":	 [ "3", "5"],
			],
		    ],
		// their contents is not important for ServiceAdjust
		"comment": $[
		    "portmap": $[],
		    "ypbind": $[],
		    "autofs": $[],
		    ],
		],
	    ],
	// Nis itself:
	"etc": $[
	    "yp_conf": $[
		"servers": [ "10.20.30.40", "10.20.30.80" ],
		"defaultbroadcast": false,
		"domainservers": $[
		    "otherdomain": [ "1.2.3.4" ],
		    ],
		"broadcast": $[
		    "otherdomain": true,
		    ],
		],
//	    "defaultdomain": "mydomain",
	    "nsswitch_conf": $[
		"passwd": "compat",
		"group": "compat",
		"hosts": "files dns6",
		"automount": "files",
		],
	    ],
	"sysconfig": $[
	    "ypbind": $[
		"YPBIND_LOCAL_ONLY": "no",
		"YPBIND_BROADCAST": "no",
		"YPBIND_BROKEN_SERVER": "no",
		"YPBIND_OPTIONS": "",
		],
	    "dhcp": dhcp_no,
	    ],
	];

    map WRITE = $[
	];

    map EXECUTE = $[
	"target": $[
	    // ok if used both for `domainname` and `rcypbind start`
	    "bash_output": $[
		"exit": 0,
		"stdout": "mydomain\n",
		"stderr": "",
		],
	    "remove": true, // /etc/yp.conf.sv
	    ],
	];

    DUMP ("DHCP off");
    TEST (``(Nis::Read ()), [READ, WRITE, EXECUTE], nil);
    TEST (``(Nis::Write ()), [READ, WRITE, EXECUTE], nil);

    DUMP ("DHCP on");
    READ["sysconfig", "dhcp"] = dhcp_yes;
    TEST (``(Nis::Read ()), [READ, WRITE, EXECUTE], nil);
    TEST (``(Nis::Write ()), [READ, WRITE, EXECUTE], nil);
}
