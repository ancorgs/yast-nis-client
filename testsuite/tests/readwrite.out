Dump	DHCP off
Read	.init.scripts.exists "ypbind" true
Read	.init.scripts.runlevel "ypbind" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.etc.yp_conf.servers ["10.20.30.40", "10.20.30.80"]
Read	.etc.yp_conf.defaultbroadcast false
Read	.etc.yp_conf.domainservers $["otherdomain":["1.2.3.4"]]
Read	.etc.yp_conf.broadcast $["otherdomain":true]
Read	.etc.yp_conf.slp $["slpdomain":true]
Execute	.target.bash_output "/bin/ypdomainname" $["exit":0, "stderr":"", "stdout":"mydomain\n"]
Read	.sysconfig.network.dhcp.DHCLIENT_MODIFY_NIS_CONF "no"
Read	.sysconfig.network.dhcp.DHCLIENT_SET_DOMAINNAME "no"
Execute	.target.bash "test -f /var/run/dhcpcd-*.pid" 0
Execute	.target.bash "/bin/grep -qs '^NISSERVERS' /var/lib/dhcpcd/dhcpcd-*.info" 0
Execute	.target.bash "/bin/grep -qs '^NISDOMAIN' /var/lib/dhcpcd/dhcpcd-*.info" 0
Read	.sysconfig.ypbind.YPBIND_LOCAL_ONLY "no"
Read	.sysconfig.ypbind.YPBIND_BROADCAST "no"
Read	.sysconfig.ypbind.YPBIND_BROKEN_SERVER "no"
Read	.sysconfig.ypbind.YPBIND_OPTIONS ""
Read	.init.scripts.exists "autofs" true
Read	.init.scripts.runlevel "autofs" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.etc.nsswitch_conf."hosts" "files dns6"
Read	.etc.nsswitch_conf."passwd" "compat"
Read	.etc.nsswitch_conf."passwd_compat" nil
Read	.sysconfig.displaymanager.DISPLAYMANAGER nil
Return	true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.runlevel "portmap" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "portmap" $["autofs":$[], "portmap":$[], "ypbind":$[]]
Read	.init.scripts.exists "ypbind" true
Read	.init.scripts.exists "ypbind" true
Read	.init.scripts.exists "ypbind" true
Read	.init.scripts.runlevel "ypbind" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "ypbind" $["autofs":$[], "portmap":$[], "ypbind":$[]]
Write	.etc.defaultdomain "mydomain" true
Execute	.target.remove "/etc/yp.conf.sv" true
Write	.etc.yp_conf.servers ["10.20.30.40", "10.20.30.80"] true
Write	.etc.yp_conf.defaultbroadcast false true
Write	.etc.yp_conf.domainservers $["otherdomain":["1.2.3.4"]] true
Write	.etc.yp_conf.broadcast $["otherdomain":true] true
Write	.etc.yp_conf.slp $["slpdomain":true] true
Write	.etc.yp_conf nil true
Write	.sysconfig.ypbind.YPBIND_LOCAL_ONLY "no" true
Write	.sysconfig.ypbind.YPBIND_BROADCAST "no" true
Write	.sysconfig.ypbind.YPBIND_BROKEN_SERVER "no" true
Write	.sysconfig.ypbind.YPBIND_OPTIONS "" true
Write	.sysconfig.ypbind nil true
Write	.sysconfig.network.dhcp.DHCLIENT_SET_DOMAINNAME "no" true
Write	.sysconfig.network.dhcp.DHCLIENT_MODIFY_NIS_CONF "no" true
Write	.sysconfig.network.dhcp nil true
Read	.etc.nsswitch_conf."passwd" "compat"
Write	.etc.nsswitch_conf."passwd_compat" nil true
Read	.etc.nsswitch_conf."group" "compat"
Write	.etc.nsswitch_conf."group_compat" nil true
Read	.etc.nsswitch_conf."shadow" nil
Write	.etc.nsswitch_conf."shadow" "compat" true
Write	.etc.nsswitch_conf."shadow_compat" nil true
Execute	.target.bash "/usr/bin/grep -q '^[+-]' /etc/passwd" 0
Execute	.target.bash "/usr/bin/grep -q '^[+-]' /etc/shadow" 0
Execute	.target.bash "/usr/bin/grep -q '^[+-]' /etc/group" 0
Write	.etc.nsswitch_conf nil true
Read	.etc.nsswitch_conf."automount" "files"
Write	.etc.nsswitch_conf."automount" "files nis" true
Write	.etc.nsswitch_conf nil true
Read	.init.scripts.exists "autofs" true
Read	.init.scripts.exists "autofs" true
Read	.init.scripts.exists "autofs" true
Read	.init.scripts.runlevel "autofs" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "autofs" $["autofs":$[], "portmap":$[], "ypbind":$[]]
Read	.init.scripts.exists "ypbind" true
Execute	.target.bash "/etc/init.d/ypbind stop" $["TERM":"raw"] 0
Read	.init.scripts.exists "portmap" true
Execute	.target.bash "/etc/init.d/portmap status" $["TERM":"raw"] 0
Execute	.target.bash_output "/etc/init.d/ypbind start" $["TERM":"raw"] $["exit":0, "stderr":"", "stdout":"mydomain\n"]
Execute	.target.bash "/usr/bin/ypwhich >/dev/null" 0
Read	.init.scripts.exists "nscd" true
Execute	.target.bash "/etc/init.d/nscd status" $["TERM":"raw"] 0
Read	.init.scripts.exists "nscd" true
Execute	.target.bash "/etc/init.d/nscd restart" $["TERM":"raw"] 0
Read	.init.scripts.exists "autofs" true
Execute	.target.bash "/etc/init.d/autofs stop" $["TERM":"raw"] 0
Read	.init.scripts.exists "autofs" true
Execute	.target.bash "/etc/init.d/autofs start" $["TERM":"raw"] 0
Return	true
Dump	DHCP on
Read	.init.scripts.exists "ypbind" true
Read	.init.scripts.runlevel "ypbind" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.etc.yp_conf.servers ["10.20.30.40", "10.20.30.80"]
Read	.etc.yp_conf.defaultbroadcast false
Read	.etc.yp_conf.domainservers $["otherdomain":["1.2.3.4"]]
Read	.etc.yp_conf.broadcast $["otherdomain":true]
Read	.etc.yp_conf.slp $["slpdomain":true]
Execute	.target.bash_output "/bin/ypdomainname" $["exit":0, "stderr":"", "stdout":"mydomain\n"]
Read	.sysconfig.network.dhcp.DHCLIENT_MODIFY_NIS_CONF "yes"
Read	.sysconfig.network.dhcp.DHCLIENT_SET_DOMAINNAME "yes"
Execute	.target.bash "test -f /var/run/dhcpcd-*.pid" 0
Execute	.target.bash "/bin/grep -qs '^NISSERVERS' /var/lib/dhcpcd/dhcpcd-*.info" 0
Execute	.target.bash "/bin/grep -qs '^NISDOMAIN' /var/lib/dhcpcd/dhcpcd-*.info" 0
Read	.sysconfig.ypbind.YPBIND_LOCAL_ONLY "no"
Read	.sysconfig.ypbind.YPBIND_BROADCAST "no"
Read	.sysconfig.ypbind.YPBIND_BROKEN_SERVER "no"
Read	.sysconfig.ypbind.YPBIND_OPTIONS ""
Read	.init.scripts.exists "autofs" true
Read	.init.scripts.runlevel "autofs" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.etc.nsswitch_conf."hosts" "files dns6"
Read	.etc.nsswitch_conf."passwd" "compat"
Read	.etc.nsswitch_conf."passwd_compat" nil
Read	.sysconfig.displaymanager.DISPLAYMANAGER nil
Return	true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.exists "portmap" true
Read	.init.scripts.runlevel "portmap" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "portmap" $["autofs":$[], "portmap":$[], "ypbind":$[]]
Read	.init.scripts.exists "ypbind" true
Read	.init.scripts.exists "ypbind" true
Read	.init.scripts.exists "ypbind" true
Read	.init.scripts.runlevel "ypbind" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "ypbind" $["autofs":$[], "portmap":$[], "ypbind":$[]]
Write	.sysconfig.ypbind.YPBIND_LOCAL_ONLY "no" true
Write	.sysconfig.ypbind.YPBIND_BROADCAST "no" true
Write	.sysconfig.ypbind.YPBIND_BROKEN_SERVER "no" true
Write	.sysconfig.ypbind.YPBIND_OPTIONS "" true
Write	.sysconfig.ypbind nil true
Write	.sysconfig.network.dhcp.DHCLIENT_SET_DOMAINNAME "yes" true
Write	.sysconfig.network.dhcp.DHCLIENT_MODIFY_NIS_CONF "yes" true
Write	.sysconfig.network.dhcp nil true
Read	.etc.nsswitch_conf."passwd" "compat"
Write	.etc.nsswitch_conf."passwd_compat" nil true
Read	.etc.nsswitch_conf."group" "compat"
Write	.etc.nsswitch_conf."group_compat" nil true
Read	.etc.nsswitch_conf."shadow" nil
Write	.etc.nsswitch_conf."shadow" "compat" true
Write	.etc.nsswitch_conf."shadow_compat" nil true
Execute	.target.bash "/usr/bin/grep -q '^[+-]' /etc/passwd" 0
Execute	.target.bash "/usr/bin/grep -q '^[+-]' /etc/shadow" 0
Execute	.target.bash "/usr/bin/grep -q '^[+-]' /etc/group" 0
Write	.etc.nsswitch_conf nil true
Read	.etc.nsswitch_conf."automount" "files"
Write	.etc.nsswitch_conf."automount" "files nis" true
Write	.etc.nsswitch_conf nil true
Read	.init.scripts.exists "autofs" true
Read	.init.scripts.exists "autofs" true
Read	.init.scripts.exists "autofs" true
Read	.init.scripts.runlevel "autofs" $["autofs":$["start":["3", "5"], "stop":["3", "5"]], "portmap":$["start":["3", "5"], "stop":["3", "5"]], "ypbind":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "autofs" $["autofs":$[], "portmap":$[], "ypbind":$[]]
Read	.init.scripts.exists "ypbind" true
Execute	.target.bash "/etc/init.d/ypbind stop" $["TERM":"raw"] 0
Read	.init.scripts.exists "portmap" true
Execute	.target.bash "/etc/init.d/portmap status" $["TERM":"raw"] 0
Execute	.target.bash_output "/etc/init.d/ypbind start" $["TERM":"raw"] $["exit":0, "stderr":"", "stdout":"mydomain\n"]
Execute	.target.bash "/usr/bin/ypwhich >/dev/null" 0
Read	.init.scripts.exists "nscd" true
Execute	.target.bash "/etc/init.d/nscd status" $["TERM":"raw"] 0
Read	.init.scripts.exists "nscd" true
Execute	.target.bash "/etc/init.d/nscd restart" $["TERM":"raw"] 0
Read	.init.scripts.exists "autofs" true
Execute	.target.bash "/etc/init.d/autofs stop" $["TERM":"raw"] 0
Read	.init.scripts.exists "autofs" true
Execute	.target.bash "/etc/init.d/autofs start" $["TERM":"raw"] 0
Return	true
